netutils/netinit/netinit.c

```diff
@@ -357,6 +367,86 @@ static void netinit_set_macaddr(void)
 #  define netinit_set_macaddr()
 #endif
 
+#ifdef CONFIG_FSUTILS_IPCFG
+/****************************************************************************
+ * Name: netinit_get_ipaddrs
+ *
+ * Description:
+ *   Setup IP addresses.
+ *
+ *   For 6LoWPAN, the IP address derives from the MAC address.  Setting it
+ *   to any user provided value is asking for trouble.
+ *
+ ****************************************************************************/
+
+static int netinit_get_ipaddrs(void)
+{
+#ifdef CONFIG_NETINIT_DHCPC
+  bool use_dhcp   = false;
+#endif
+  bool use_static = false;
+#ifdef CONFIG_NET_IPv4
+  struct in_addr addr;
+  int ret;
+
+  ret = ipcfg_read(NET_DEVNAME, &g_netconf);
+  if (ret < 0)
+    {
+      return ret;
```

Comment:
can we initialize g_netconf here with the default value? so we don't need duplicate netlib_set_xxx.

Reply:
No. That can be done when we add IPv6.

Reply:
It's not related to IPv6 even IPv4 we have the duplicaed code from line 393-403 and line 468-478.

Reply:
It is not the same 

![image](https://user-images.githubusercontent.com/1945821/94931423-978d2e80-047c-11eb-9a70-f50b4792c579.png)

![image](https://user-images.githubusercontent.com/1945821/94931469-a673e100-047c-11eb-860a-c1d9c091174c.png)

the compile time #ifdefs will make it not readable just setting the protocol.

Reply:
Do you think the follow code is reasonable and more simple?
```
ret = ipcfg_read(NET_DEVNAME, &g_netconf);
if (ret < 0)
   {
     /* Set the default value from Kconfig here */
     g_netconf.proto = BOOTPROTO_STATIC;
     g_netconf.s_addr = CONFIG_NETINIT_IPADDR;
     ...
   }

/* Share the same code regardless where we get the config from EEPROM or Kconfig */
...
```

Reply:
Will it compile if DCP is selected. CONFIG_NETINIT_IPADDR is not defined

Reply:
The code just give my rough idea to you, please adjust as need. The keypoint is that we can initialzie g_netconf with the default value if  ipcfg_read return fail, then we don't duplicate netlib_set_xxx twice.

Reply:
I will not be changing this. 1) It is stable, 2) easy to read, 3) compiles with all the permutations of the CONFIG settings. 

I suspect this code will change when the V6 part is done. Look to this is a major code saving you can add with all the IPv6.

Reply:
> I will not be changing this. 1) It is stable, 2) easy to read, 3) compiles with all the permutations of the CONFIG settings.
> 
> I suspect this code will change when the V6 part is done. Look to this is a major code saving you can add with all the IPv6.

I don't agree, let's make the code in a good shape in the first place. If you code get merged as the current state, it will encourage people who add IPv6 support duplicate the similar code again.